aspnet/mvc/views/partial.rst

```diff
@@ -1,8 +1,128 @@
+  :lines: 9
+
+You can also render partial views asynchronously using ``PartialAsync``:
+
+.. literalinclude:: partial/sample/src/PartialViewsSample/Views/Home/About.cshtml
+  :lines: 8
+
+A third way to render partial views is to call directly to ``RenderPartial``. This method doesn't return a result, it streams the rendered output directly to the response. Because it doesn't return a result, it must be called within a razor code block. You can also call its async equivalent, as shown here:
+
+.. literalinclude:: partial/sample/src/PartialViewsSample/Views/Home/About.cshtml
+  :lines: 10-13
+
+Because it streams the result directly, ``RenderPartial``/``RenderPartialAsync`` may perform better in some scenarios. However, in most cases it's recommended you use ``Render`` or ``RenderAsync``.
+
+.. note:: If your views need to execute code, the recommended pattern is to use a :doc:`view component <view-components>` instead of a partial view.
+
+Partial View Discovery
+^^^^^^^^^^^^^^^^^^^^^^
+
+When referencing a partial view, you can refer to its location in several ways:
+
+.. code-block:: text
+
+  // uses a view in current folder with this name
+  // if none is found, searches the Shared folder
+  @Html.Partial("ViewName")
+
+  // a view with this name must be in the same folder
+  @Html.Partial("ViewName.cshtml")
+  
+  // locate the view based on the application root
+  // Paths start with "/" or "~/" refer to same location
+  @Html.Partial("~/Views/Folder/ViewName.cshtml")
+  @Html.Partial("/Views/Folder/ViewName.cshtml")
+  
+  // locate the view using relative paths
+  @Html.Partial("../Account/LoginPartial.cshtml")
+
+If desired, you can have different partial views with the same name in different view folders. When referencing the views by name (without file extension), views in each folder will use the partial view in the same folder with them. You can also specify a default partial view to use, placing it in the *Shared* folder. This view will be used by any views that don't have their own copy of the partial view in their folder.
+
+Partial views can be *chained*. That is, you can call from one partial view to another, so long as you don't create a loop. Within each view or partial view, relative paths are always relative to that view, not the root or parent view.
+
+.. note:: If you declare a :doc:`razor <razor>` ``section`` in a partial view, it will not be visible to its parent(s); it will be limited to the partial view.
+
+Accessing Data From Partial Views
+---------------------------------
+
+When a partial view is instantiated, it is given its own copy of the parent view's ``ViewData`` collection. This provides the partial view with access to the parent view's data. However, updates made to the data within the partial view are not shared with the parent view - its data remains unchanged. If desired, you can pass a custom ``ViewDataDictionary`` to the partial view:
```

Comment:
No need to create a "custom `ViewDataDictionary`" if a partial view needs to communicate back to the view that called `Html.PartialAsync()`. For example, could use a view model containing whatever information is needed. Could also have the calling view place a complex object in a well-known `ViewData` location and the partial view update properties of that object. But in general communication back "up" is rare and tends to make the partial view less generally usable.